### ISSUE TITLE
problem about run d4rl_bcq.py

### ISSUE THREAD

**Issue by GongYanfu** [aai-institute/tianshou#873]
- [x] I have marked all applicable categories:
    + [x] exception-raising bug
    + [x] RL algorithm bug
    + [x] documentation request (i.e. "X is missing from the documentation.")
    + [x] new feature request
- [x] I have visited the [source website](https://github.com/thu-ml/tianshou/)
- [x] I have searched through the [issue tracker](https://github.com/thu-ml/tianshou/issues) for duplicates
- [x] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import tianshou, gymnasium as gym, torch, numpy, sys
  print(tianshou.__version__, gym.__version__, torch.__version__, numpy.__version__, sys.version, sys.platform)
  ```
![image](https://github.com/thu-ml/tianshou/assets/46698455/fb409a00-36a4-4f09-8caf-e96a53cb5055)

I used the commandas bellow:
$ nohup python -u offline/d4rl_bcq.py --task HalfCheetah-v2 --expert-data-task halfcheetah-expert-v2 > offline/logs/train/HalfCheetah-v2/halfcheetah-expert-v2.log 2>&1 &
but problem as belloe occured: 

![image](https://github.com/thu-ml/tianshou/assets/46698455/613d5722-a39d-48f9-9548-0fba577b509c)

**Comment by Trinkle23897** [comment]
Have you installed d4rl dataset package? If yes, try to import it at beginning of the script

**Comment by GongYanfu** [comment]
After I changed ther command like bellow, It occurs another error:
![image](https://github.com/thu-ml/tianshou/assets/46698455/e346d37e-5da7-4fdd-b374-0b00cace84f6)

**Comment by GongYanfu** [comment]
> Have you installed d4rl dataset package? If yes, try to import it at beginning of the script

d4rl dataset package?
do you mean that I need install d4rl?

**Comment by Trinkle23897** [comment]
https://github.com/thu-ml/tianshou/blob/f8808d236f53d86583c74f459365c9cc4184256c/examples/offline/README.md?plain=1#L5-L9

**Comment by thkkk** [comment]
I think the problem may lie in the compatibility of d4rl and gymnasium.  We implemented these offline RL algorithm when the gym was not "gymnasium". So a simple way to solve this problem could be to modify the `d4rl` dataset loading process, in https://github.com/thu-ml/tianshou/blob/master/examples/offline/utils.py
```python
import gymnasium as gym
```
change it to
```python
import gym
````

**Comment by MischaPanch** [comment]
d4rl is discontinued in favor of [minari](https://github.com/Farama-Foundation/Minari). It probably makes sense to switch all offline RL code in tianshou to minary as well

**Comment by MischaPanch** [comment]
Will be addressed in #932, closing this
